lib/kube/proxy/forwarder.go

```diff
@@ -654,15 +657,19 @@ func (f *Forwarder) getKubeGroupsAndUsers(
 			// Get list of allowed kube user/groups based on kubernetes service labels.
 			labels := types.CombineLabels(c.StaticLabels, c.DynamicLabels)
 			labelsMatcher := services.NewKubernetesClusterLabelMatcher(labels)
-			groups, users, err = roles.CheckKubeGroupsAndUsers(sessionTTL, false, labelsMatcher)
+			kubePerms, err = roles.CheckKubeGroupsAndUsers(sessionTTL, false, labelsMatcher)
 			if err != nil {
-				return nil, nil, trace.Wrap(err)
+				return services.KubePermissions{}, trace.Wrap(err)
 			}
-			return groups, users, nil
+			return kubePerms, nil
 		}
 	}
+
 	// kubeClusterName not found. Empty list of allowed kube users/groups is returned.
-	return []string{}, []string{}, nil
+	return services.KubePermissions{
+		Groups: make([]string, 0),
+		Users:  make([]string, 0),
+	}, nil
 }
```

Comment:
Nit: Just returning `services.KubePermissions{}` should be fine with nil slices.